### PR TITLE
Fix regressions in config scaffolding, add list / float support

### PR DESCRIPTION
**Pull request by bengotow**
This PR also adds a `console.warn` which should make this easier to debug / triage if it ever breaks in the future. I'll see if we can get some jest tests added for this functionality soon.

Related: #820

### CODE REVIEW

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 1]
```diff
@@ -131,40 +131,50 @@ export function scaffoldConfig(pipeline: ConfigEditorPipelineFragment): string {
...
-    if (placeholders[typeName]) {
-      return placeholders[typeName];
+  const configPlaceholderFor = (typeKey: string, commentDepth: number): any => {
+    if (placeholders[typeKey] !== undefined) {
+      return placeholders[typeKey];
     }
 
-    const type = pipeline.configTypes.find(t => t.name === typeName);
-    if (!type || type.__typename !== "CompositeConfigType") return null;
+    if (typeKey.startsWith("List.")) {
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: let's not do substr matching here and check the explicit isList property or GraphQL type name

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 2]
```diff
@@ -131,40 +131,50 @@ export function scaffoldConfig(pipeline: ConfigEditorPipelineFragment): string {
...
+        typeKey.substr(5),
+        commentDepth
+      );
+      if (innerPlaceholder === null) return null;
+      return [innerPlaceholder];
+    }
 
+    const type = pipeline.configTypes.find(t => t.key === typeKey);
+    if (!type || type.__typename !== "CompositeConfigType") {
+      console.warn(
```
Comment (mgasner) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: 👍

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 3]
```diff
@@ -131,40 +131,50 @@ export function scaffoldConfig(pipeline: ConfigEditorPipelineFragment): string {
...
-        }
-      });
+    type.fields.forEach((field, idx) => {
+      const startComment = type.isSelector && idx > 0;
+      const fieldCommentDepth =
+        commentDepth > 0 ? commentDepth + 2 : startComment ? 1 : 0;
+
+      const val = configPlaceholderFor(field.configType.key, fieldCommentDepth);
+      if (
+        val === null ||
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: seems like we have a utility to capture this "is_nullish" state

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 4]
```diff
@@ -131,40 +131,50 @@ export function scaffoldConfig(pipeline: ConfigEditorPipelineFragment): string {
...
+        (val instanceof Object && Object.keys(val).length == 0)
+      ) {
+        return;
+      }
+      if (fieldCommentDepth > 0) {
+        result[`COMMENTED_${fieldCommentDepth}_${field.name}`] = val;
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: can you make a comment re: what this synthesized key represents?

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 5]
```diff
@@ -131,40 +131,50 @@ export function scaffoldConfig(pipeline: ConfigEditorPipelineFragment): string {
...
-      return placeholders[typeName];
+  const configPlaceholderFor = (typeKey: string, commentDepth: number): any => {
+    if (placeholders[typeKey] !== undefined) {
+      return placeholders[typeKey];
     }
 
-    const type = pipeline.configTypes.find(t => t.name === typeName);
-    if (!type || type.__typename !== "CompositeConfigType") return null;
+    if (typeKey.startsWith("List.")) {
+      const innerPlaceholder = configPlaceholderFor(
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: Ah I see. Can you create a task to define a GraphQL interface for what you want here in terms of accessing the inner type names without having to do substring chicanery?

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx (hunk 1 above)
Comment (bengotow) [reply js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: Ahh good call I forgot we had an isList 👍

**Comment by bengotow** [comment]
Hey @schrockn I replaced this with an implementation that performs innerTypes resolution recursively to handle lists without just chopping of `List.`

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 6]
```diff
@@ -126,46 +140,83 @@ export async function checkConfig(
...
 
-        const val = configPlaceholderFor(
-          field.configType.key,
-          fieldCommentDepth
-        );
-        if (!val || Object.keys(val).length == 0) return;
+        let val = null;
+
+        if (field.configType.__typename === "ListConfigType") {
+          let inner: ConfigEditorPipelineFragment_configTypes_CompositeConfigType_fields_configType_ListConfigType_innerTypes =
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: oh man can we alias that one somehow

File: js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx [hunk 7]
```diff
@@ -126,46 +140,83 @@ export async function checkConfig(
...
+      console.warn(`Unsure of how to scaffold missing type: ${typeKey}`);
+      return null;
+    }
 
-    const result = {};
-    type.fields
-      .filter(f => !f.isOptional)
-      .forEach((field, idx) => {
+    if (type.__typename === "CompositeConfigType") {
+      const result = {};
```
Comment (schrockn) [inline js_modules/dagit/src/configeditor/ConfigEditorUtils.tsx]: i find const pretty misleading in these cases. (aside: const was a mistake!)

**Pull request merged**